src/Microsoft.ML.Transforms/TextCatalog.cs

```diff
@@ -1,277 +0,0 @@
-        /// <param name="mhstep">Number of Metropolis Hasting step.</param>
-        /// <param name="numIterations">Number of iterations.</param>
-        /// <param name="likelihoodInterval">Compute log likelihood over local dataset on this iteration interval.</param>
-        /// <param name="numThreads">The number of training threads. Default value depends on number of logical processors.</param>
-        /// <param name="numMaxDocToken">The threshold of maximum count of tokens per doc.</param>
-        /// <param name="numSummaryTermPerTopic">The number of words to summarize the topic.</param>
-        /// <param name="numBurninIterations">The number of burn-in iterations.</param>
-        /// <param name="resetRandomGenerator">Reset the random number generator for each document.</param>
-        public static LatentDirichletAllocationEstimator LatentDirichletAllocation(this TransformsCatalog.TextTransforms catalog,
-            string inputColumn,
-            string outputColumn = null,
-            int numTopic = LatentDirichletAllocationEstimator.Defaults.NumTopic,
-            float alphaSum = LatentDirichletAllocationEstimator.Defaults.AlphaSum,
-            float beta = LatentDirichletAllocationEstimator.Defaults.Beta,
-            int mhstep = LatentDirichletAllocationEstimator.Defaults.Mhstep,
-            int numIterations = LatentDirichletAllocationEstimator.Defaults.NumIterations,
-            int likelihoodInterval = LatentDirichletAllocationEstimator.Defaults.LikelihoodInterval,
-            int numThreads = LatentDirichletAllocationEstimator.Defaults.NumThreads,
-            int numMaxDocToken = LatentDirichletAllocationEstimator.Defaults.NumMaxDocToken,
-            int numSummaryTermPerTopic = LatentDirichletAllocationEstimator.Defaults.NumSummaryTermPerTopic,
-            int numBurninIterations = LatentDirichletAllocationEstimator.Defaults.NumBurninIterations,
-            bool resetRandomGenerator = LatentDirichletAllocationEstimator.Defaults.ResetRandomGenerator)
-            => new LatentDirichletAllocationEstimator(CatalogUtils.GetEnvironment(catalog), inputColumn, outputColumn, numTopic, alphaSum, beta, mhstep, numIterations, likelihoodInterval, numThreads, numMaxDocToken,
-                numSummaryTermPerTopic, numBurninIterations, resetRandomGenerator);
-
-        /// <summary>
-        /// Uses <a href="https://arxiv.org/abs/1412.1576">LightLDA</a> to transform a document (represented as a vector of floats)
-        /// into a vector of floats over a set of topics.
-        /// </summary>
-        /// <param name="catalog">The transform's catalog.</param>
-        /// <param name="columns"> Describes the parameters of LDA for each column pair.</param>
-        public static LatentDirichletAllocationEstimator LatentDirichletAllocation(this TransformsCatalog.TextTransforms catalog, params LatentDirichletAllocationTransformer.ColumnInfo[] columns)
```

Comment:
>LatentDirichletAllocationEstimator [](start = 22, length = 34)

I think Abhishek  moved LDA to separate file yesterday, can you merge and wipe it as well?

Reply:
this is post Abishek's merge. By the way, do we want LDA in text or projections?

---
In reply to: [235500231](https://github.com/dotnet/machinelearning/pull/1682#discussion_r235500231) [](ancestors = 235500231)

Reply:
Sorry, got confused between catalog and static extensions. LDA doesn't make much sense outside of text, we use terminology as words, topics, etc. so Text.

---
In reply to: [235501427](https://github.com/dotnet/machinelearning/pull/1682#discussion_r235501427) [](ancestors = 235501427,235500231)